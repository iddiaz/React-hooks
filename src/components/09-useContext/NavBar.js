import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
   return (
      // <nav>
      //    <ul>
      //       <li>
      //         <Link to="/">Home</Link>
      //       </li>
      //       <li>
      //         <Link to="/about">About</Link>
      //       </li>
      //       <li>
      //         <Link to="/login">Login</Link>
      //       </li>
          
      //    </ul>         
      // </nav>

      <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <div className="container-fluid">
        <NavLink className="navbar-brand" to="/">
          useContext
        </NavLink>
      
        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
          <div className="navbar-nav">
            <NavLink className="nav-link " aria-current="page" to="/">
              Home
            </NavLink>
            <NavLink className="nav-link" to="about">
              About
            </NavLink>
            <NavLink className="nav-link" to="login">
              Login
            </NavLink>
            
          </div>
        </div>
      </div>
    </nav>


   )
}
