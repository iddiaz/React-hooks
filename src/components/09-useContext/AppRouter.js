import React from 'react';
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";

import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
   return (
      <Router>
         <div>

            <NavBar />

            <div className="container">

            <Routes>
               <Route path="/" element={ <HomeScreen /> } />            
               <Route path="/about" element={ <AboutScreen /> } />
               <Route path="/login" element={ <LoginScreen /> } />
               <Route path="*" element={ <HomeScreen /> } />
            </Routes>
            </div>

         </div>
      </Router>
   )
}
