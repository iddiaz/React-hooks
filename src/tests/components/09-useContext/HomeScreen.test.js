import { React } from 'react';
import { shallow, mount } from 'enzyme';
import { HomeScreen } from './../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Pruebas componente <HomeScreen />', ()=>{
   
   const user = {
      name: 'Ivan',
      email: 'email@email.com'
   }
   const wrapper = mount(
      <UserContext.Provider value={{
         user
      }}>
         <HomeScreen /> 
      </UserContext.Provider>
      );

   test('Debe mostrarse correctamente', () => {

      expect( wrapper).toMatchSnapshot();

   }) 


})