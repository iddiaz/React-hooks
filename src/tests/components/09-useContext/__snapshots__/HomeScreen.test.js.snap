// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente <HomeScreen /> Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre
    className="container"
  >
    {
   "name": "Ivan",
   "email": "email@email.com"
}
  </pre>
</div>
`;
